stm32-modules/include/thermocycler-refresh/firmware/motor_hardware.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __MOTOR_HARDWARE_H
+#define __MOTOR_HARDWARE_H
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// ----------------------------------------------------------------------------
+// Type definitions
+
+// This structure is used to define callbacks out of motor interrupts
+typedef struct {
+    void (*lid_stepper_complete)();
```

Comment:
[nit] due to an extreme case of Hating Function Pointer Syntax, I hate function pointer syntax. Can we typedef this?